src/main/java/duke/command/CommandSchedule.java

```diff
@@ -0,0 +1,58 @@
+package duke.command;
+
+import duke.task.TaskList;
+import duke.worker.Parser;
+import duke.worker.Ui;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class CommandSchedule extends Command {
+    protected String userInput;
+
+
+    //constructor
+    public CommandSchedule(String userInput) {
+        this.userInput = userInput;
+        this.commandType = CommandType.VIEWSCHEDULE;
+    }
+
+
+    @Override
+    public void execute(TaskList taskList) {
+        String dateInput = Parser.removeStr(this.commandType.toString(), this.userInput);
+        Ui.dukeSays("Here is your schedule for the following date '"
+                + dateInput
+                + "'."
+        );
+        try {
+            printSchedule(dateInput, taskList);
+        } catch (Exception e) {
+            System.out.println("Read invalid input");
+        }
+    }
+
+    private void printSchedule(String dateInput, TaskList taskList) throws ParseException {
+        Date date = new Date();
+        //insert code to convert string to date type
+        SimpleDateFormat format = new SimpleDateFormat("dd/MM/yyyy");
+        Date userDate = format.parse(dateInput);
+        userDate.setTime(0);
+        for (int index = 0; index < taskList.getSize(); ++index) {
+            try {
```

Comment:
Can abstract this further. I believe in terms of code quality, the Prof mentioned maximum 2-3 levels before you need to abstract.